addons/web_tour/static/src/tour_service/tour_service.js

```diff
@@ -288,7 +289,21 @@ export const tourService = {
             const macro = convertToMacro(tour, pointer, options);
             const willUnload = callWithUnloadCheck(() => {
                 if (tour.url && tour.url !== options.startUrl && options.redirect) {
-                    window.location.href = window.location.origin + tour.url;
+                    const fromUrl = new URL(browser.location.href);
+                    const toUrl = new URL(browser.location.origin + tour.url);
+                    const debugMode = fromUrl.searchParams.get("debug");
+                    if (debugMode) {
+                        const allDebug = [];
+                        [toUrl.searchParams.get("debug") || "", debugMode].forEach((string) => {
+                            string.split(",").forEach((mode) => {
+                                if (mode && !allDebug.includes(mode)) {
+                                    allDebug.push(mode);
+                                }
+                            });
+                        });
+                        toUrl.searchParams.set("debug", sortBy(allDebug).join(","));
+                    }
```

Comment:
```suggestion
                    if (debugMode) {
                        const allDebug = new Set([
                            ...toUrl.searchParams.get("debug").split(","),
                            ...debugMode.split(","),
                        ]);
                        toUrl.searchParams.set("debug", [...allDebug].join(","));
                    }
```

Reply:
Yep, I was imitating what you discovered: for small enough arrays, checking for presence in O(n) is faster than the overhead of creating a Set and checking presence in O(1)

Reply:
This is not hot code :sweat_smile: I'd definitely default to more readable code unless there's a strong reason to believe the performance benefit is significant.